### DISCUSSION TITLE
dev server freezes/crashes, reason unclear (edited)

### DISCUSSION THREAD

**Discussion by 205g0** [sveltejs/kit#9340] [Q&A]
Edit: see also comment below, crashes come erratically and hard to track down

Pretty much the title, if you use e.g. `localStorage` outside a `browser` check with `ssr = true` in a .ts file let the dev server silently freezes. Which makes sense but I wonder if there could come some warning and the possibilty to continue without the need to restart the server (even after fixing the bug).

**Comment by 205g0** [comment]
So, I get this dev server freezing more often now and I can't find the cause. I checked for usage of browser APIs but it's still happening. Always after saving changes to `.ts` files and when `ssr = true`. Super strange and tbh, somehow annoying. If there was at least some hint, what went wrong and why the server is crashing...

**Comment by hanszoons** [comment]
About the unexpected and unexplainable freezing, I raised this issue in the Vite repo some time ago, which is still open : https://github.com/vitejs/vite/issues/11468 

There are some suggestions there (but none of them worked for me).

**Reply by 205g0** [reply]
Thanks for the link, I skimmed through it and I have exactly what you've described over there:

_Just tried it again, same predictable hang when I edit an imported function._

But I can't find any indication on which imported function edits this happens.
